server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java

```diff
@@ -137,7 +137,9 @@ public ReadOnlyEngine(
                 }
                 if (seqNoStats == null) {
                     seqNoStats = buildSeqNoStats(config, lastCommittedSegmentInfos);
-                    ensureMaxSeqNoEqualsToGlobalCheckpoint(seqNoStats);
```

Comment:
Nit - would it make more readable this code, if we offload this check within `ensureMaxSeqNoEqualsToGlobalCheckpoint` method -

```
 if (requireCompleteHistory == false || engineConfig.getIndexSettings().isRemoteTranslogStoreEnabled()) {
            return;
        }
```

https://github.com/opensearch-project/OpenSearch/blob/5f89081d5256aadd1d90e8618deecb63fceb473a/server/src/main/java/org/opensearch/index/engine/ReadOnlyEngine.java#L179-L181